### PR TITLE
[bitnami/kibana] Major release: Kibana 8 + Chart standardisation

### PR DESCRIPTION
**Pull request by migruiz4**
Signed-off-by: Miguel Ruiz <miruiz@vmware.com>

**Description of the change**

Updates Kibana chart to version 8 and applies standardization changes.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kibana/templates/deployment.yaml [hunk 1]
```diff
@@ -110,14 +147,19 @@ spec:
...
             {{- if .Values.elasticsearch.security.auth.enabled }}
-            - name: KIBANA_USERNAME
-              value: {{ .Values.elasticsearch.security.auth.kibanaUsername | quote }}
             - name: KIBANA_PASSWORD
               valueFrom:
                 secretKeyRef:
                   name: {{ include "kibana.elasticsearch.auth.secretName" . }}
                   key: kibana-password
             {{- end }}
+            {{- if and .Values.elasticsearch.security.auth.enabled .Values.elasticsearch.security.auth.createSystemUser }}
```
Comment (FraPazGal) [inline bitnami/kibana/templates/deployment.yaml:156]: I think I'm missing some context here. If I understood correctly, by default we have a `kibana_system` user without a password. If we enable ES's `security.auth` said user will use the password present obtained from `security.auth.kibanaPassword` or the secret at `security.auth.existingSecret`. 

What happens when this conditional is `true`? Does ES create a new `kibana_system` that uses `security.auth.elasticsearchPasswordSecret` as a password? I'm not seeing the need to have the `KIBANA_PASSWORD` env set twice with different secrets.

**Review by FraPazGal — COMMENTED** [review]
Thanks for the PR! Could you please check my comments?

File: bitnami/kibana/templates/deployment.yaml [hunk 2]
```diff
@@ -3,11 +3,18 @@ apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
 kind: Deployment
 metadata:
   name: {{ include "common.names.fullname" . }}
+  namespace: {{ template "common.names.namespace" . }}
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/kibana/templates/deployment.yaml:6]: Hmmm, I think we should instead add the `| quote` inside the helper, right?

File: bitnami/kibana/templates/service.yaml [hunk 3]
```diff
@@ -2,40 +2,51 @@ apiVersion: v1
...
-{{- if or (and .Values.metrics.enabled .Values.metrics.service.annotations) .Values.service.annotations }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.service.labels }}
+    {{- include "common.tplvalues.render" (dict "value" .Values.service.labels "context" $) | nindent 4 }}
+    {{- end }}
   annotations:
-  {{- if and .Values.metrics.enabled .Values.metrics.service.annotations }}
-    {{- tpl (toYaml .Values.metrics.service.annotations) $ | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/kibana/templates/service.yaml:12]: Nice catch! I accidentally removed the `metrics.annotations` when adding the `service.annotations`.

File: bitnami/kibana/templates/deployment.yaml [hunk 4]
```diff
@@ -46,43 +53,73 @@ spec:
...
-          image: "{{ template "kibana.volumePermissions.image" . }}"
+          image: {{ include "kibana.volumePermissions.image" . }}
           imagePullPolicy: {{ default "" .Values.volumePermissions.image.pullPolicy | quote }}
-          command: ["chown", "-R", "{{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }}", "/bitnami/kibana"]
+          command:
+            - /bin/bash
+          args:
+            - -ec
+            - |
+              mkdir -p /bitnami/kibana
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/kibana/templates/deployment.yaml:82]: I'm not sure in which cases would it be necessary, but I have seen this used in many other charts and as far as I know, it won't have any effect if the directory already exists

**Review by fmulero — DISMISSED** [review]
LGTM

File: bitnami/kibana/templates/deployment.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (FraPazGal) [reply bitnami/kibana/templates/deployment.yaml:82]: Indeed, there is no downside to leaving this. Let's keep it there.

File: bitnami/kibana/templates/deployment.yaml (hunk 1 above)
Comment (FraPazGal) [reply bitnami/kibana/templates/deployment.yaml:156]: Thanks for the explanation! The difference here is whether we are creating the `kibna_system` user with the same credentials used in ES, right?

**Review by FraPazGal — DISMISSED** [review]
LGTM

**Pull request merged**